File: Sorting/InsertionSort/Python/Insertion-Sort.py
def InsertionSort (A, n):
    for i in range (1, n):
        key = A [i]
        j = i - 1
        
        while j >= 0 and A [j] > key:
            A [j + 1] = A [j]
            j -= 1

        A [j + 1] = key

File: Sorting/InsertionSort/Java/InsertionSort.java
package Sorting.InsertionSort.Java;

public class InsertionSort {
    public static void insertionSort(int[] A, int n) {
        for (int i = 1; i < n; i++) {
            int key = A[i];
            int j = i - 1;

            while (j >= 0 && A[j] > key) {
                A[j + 1] = A[j];
                j--;
            }

            A[j + 1] = key;
        }
    }
}